sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs

```diff
@@ -68,38 +68,47 @@ internal static class TrackOneComparer
                     return false;
                 }
 
-                // Verify that the system properties contain the same elements, assuming they are non-null.
-
-                if (trackOneEvent.SystemProperties?.Count != trackTwoEvent.SystemProperties?.Count)
-                {
-                    return false;
-                }
-
                 foreach (var property in trackOneEvent.SystemProperties)
                 {
                     if (property.Key == TrackOne.ClientConstants.EnqueuedTimeUtcName)
                     {
-                        // It's necessary to be extra careful with EnqueuedTime property because its type
-                        // differs between Track One and Track Two.
-
                         var trackOneDate = (DateTime)trackOneEvent.SystemProperties[property.Key];
-                        var trackTwoDate = (DateTimeOffset)trackTwoEvent.SystemProperties[property.Key];
+                        var trackTwoDate = trackTwoEvent.SystemProperties.EnqueuedTime;
```

Comment:
What if `SystemProperties` is null? I think we should make it a special case before these properties checks.

Reply:
That's fair.   We'd blow up now, though the result would be correct, the reason would be unclear.   I'll tweak and add a couple of additional test cases.

Reply:
oh.... look at lines 63-66.    We already make sure that either both are null or both are non-null.   If we have a mixed case, we'll never hit this path.

Reply:
You're right.

Reply:
If it makes you feel any better, I didn't realize it until I went in to fix this up and was confused why I missed it earlier.  :)